File: python/username_validator.py
class UsernameValidator:
    def is_valid(self, username: str) -> bool:
        if username is None or username == "":
            return False
        if len(username) < 3 or len(username) > 20:
            return False
        if not username[0].isalpha():
            return False
        for ch in username[1:]:
            if not (ch.isalpha() or ch.isdigit() or ch == '_'):
                return False
        return True

File: cpp/username_validator.cpp
#include "username_validator.h"
#include <cctype>

bool UsernameValidator::isValid(const std::string& username) {
    if (username.empty()) {
        return false;
    }
    std::size_t len = username.size();
    if (len < 3 || len > 20) {
        return false;
    }
    if (!std::isalpha(static_cast<unsigned char>(username[0]))) {
        return false;
    }
    for (std::size_t i = 1; i < len; ++i) {
        unsigned char ch = static_cast<unsigned char>(username[i]);
        if (!std::isalpha(ch) && !std::isdigit(ch) && ch != '_') {
            return false;
        }
    }
    return true;
}
